Automattic/wp-calypso
# Improve Customizer Accessibility

When you go to the customizer, focus does not move into the customizer. This means:
1. Keyboard and Screen Reader users are not placed in the correct area (the Customizer)
2. They stay on the "previous" page of the calypso sidebar. This sidebar is not hidden, so you can still tab around and interact with the window "behind" the customizer, as the customizer is an overlay on top of the page.

![customizer focus management](https://user-images.githubusercontent.com/967608/89439535-f39f3500-d70f-11ea-88b3-375686bbd1ce.gif)

#### Changes proposed in this Pull Request

* Move focus into customizer iframe on customizer load using a `ref` one the iframe.
* Add `display: none;` to the `header` and `sidebar__navigation` on the page "behind" the customizer. This blocks access to those areas, while still allowing access to the chat, if open.
* Unrelated, but to fix a linting error, I had to remove node `url` usage and replace it with `lib/url` `getUrlParts`

I initially looked at a focus lock that would trap focus within the customizer, but that would have:
- Been a real focus trap (in a very bad way), as escape does not exit the customizer and you can't `tab` off of it to the rest of the browser window outside of the page (to change tabs, etc)
- Not allowed `tab` access to the happy chat sidebar

#### Testing instructions
* Go to  `/home/yoursite.wordpress.com` in Calypso
* Click the Customizer
* Focus should be within the customizer. Press `tab` to see where you are
* Shift + Tab should leave the page window and bring you into your browser controls in short order, not bringing you into the Calypso page menus behind the customizer
* Tab a bunch of times should bring you through to the end of the customizer page preview. After the last link, you should also exit the page and go into the browser controls without entering into the Calypso page menus behind the customizer
* Closing the customizer should bring you back into the Calypso `/home` and focus should be at the top of the page.

Improves on https://github.com/Automattic/wp-calypso/issues/43504, initial plan and info available at https://github.com/Automattic/wp-calypso/issues/43504#issuecomment-666676582
